src/handlers/gestures/panGesture.ts

```diff
@@ -0,0 +1,88 @@
+import { UnwrappedGestureHandlerEvent } from '../gestureHandlerCommon';
+import { BaseGesture, BaseGestureConfig } from './gesture';
+import {
+  PanGestureConfig,
+  PanGestureHandlerEventPayload,
+} from '../PanGestureHandler';
+
+export class PanGesture extends BaseGesture<PanGestureHandlerEventPayload> {
+  public config: BaseGestureConfig & PanGestureConfig = {};
+
+  constructor() {
+    super();
+
+    this.handlerName = 'PanGestureHandler';
+  }
+
+  setOnUpdate(
+    callback: (
+      event: UnwrappedGestureHandlerEvent<PanGestureHandlerEventPayload>
+    ) => void
+  ) {
+    this.handlers.onUpdate = callback;
+    return this;
+  }
+
+  setActiveOffsetY(offset: number | number[]) {
+    if (Array.isArray(offset)) {
+      this.config.activeOffsetYStart = offset[0];
+      this.config.activeOffsetYEnd = offset[1];
+    } else if (offset < 0) {
+      this.config.activeOffsetYStart = offset;
+    } else {
+      this.config.activeOffsetYEnd = offset;
+    }
```

Comment:
I think we can extract those methods to something shared between Gestures and GestureHandlers but it's not that important right now.